src/couch/src/couch_util.erl

```diff
@@ -228,8 +228,17 @@ get_value(Key, List, Default) ->
             Default
     end.
 
-set_value(Key, List, Value) ->
-    lists:keyreplace(Key, 1, List, {Key, Value}).
+% insert or update a {Key, Value} tuple in a list of tuples
+-spec set_value(Key, TupleList1, Value) -> TupleList2 when
+    Key :: term(),
+    TupleList1 :: [Tuple],
+    Value :: any(),
+    TupleList2 :: [Tuple].
```

Comment:
I think we're missing a `Tuple :: tuple()` or just use the `tuple()` type? 

Does it matter that it's `term()` vs `any()` for Key and Value. Maybe make them both `term()`?